torchgeo/trainers/change.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for change detection."""
+
+import os
+from typing import Any
+
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+from einops import rearrange
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.classification import (
+    BinaryAccuracy,
+    BinaryF1Score,
+    BinaryJaccardIndex,
+)
+from torchvision.models._api import WeightsEnum
+
+from ..models import FCSiamConc, FCSiamDiff, get_weight
+from . import utils
+from .base import BaseTask
+
+
+class ChangeDetectionTask(BaseTask):
+    """Change Detection. Currently supports binary change between two timesteps."""
+
+    def __init__(
+        self,
+        model: str = 'unet',
```

Comment:
Not all str are valid:
```suggestion
        model: Literal['unet', 'fcsiamdiff', 'fcsiamconc'] = 'unet',
```
where `Literal` is from `typing`